flink-runtime/src/main/java/org/apache/flink/runtime/operators/coordination/OperatorCoordinatorHolder.java

```diff
@@ -264,11 +266,12 @@ private void checkpointCoordinatorInternal(
                             result.completeExceptionally(e);
                         }
                     }
-                });
+                },
+                mainThreadExecutor);
```

Comment:
I guess javadoc section `Concurrency and Threading Model` should change a bit.

Reply:
yes, indeed, it needs to be adjusted, thanks!